src/core/json-schema/src/formly-json-schema.service.ts

```diff
@@ -426,6 +428,10 @@ export class FormlyJsonschema {
       }
     }
 
+    if (type === 'string' && format !== undefined) {
+      return format;
```

Comment:
We should restrict this to the date & time only and check  if the type is defined else we should fallback to the Native type by setting `type` to the `templateOptions`:

```ts
{
  type: 'string',
  templateOptions: {
    type: 'date', // 'time' | 'datetime',
  }
}
```

Reply:
So would changing the implementation to something like this be sufficient?

```typescript
if (type === 'string' && (format === 'date' || format === 'time')) {
  return format;
}
```

I don't quite understand your point about the `templateOptions`. Should they be changed as well?